bin/fetch.rs

```diff
@@ -91,11 +97,14 @@ fn main() -> Result<(), Box<dyn Error>> {
         state.insert(address, account);
     }
 
-    // Write bytecodes to disk.
-    let file_bytecodes = File::create("data/bytecodes.bincode")
-        .map_err(|err| format!("Failed to create bytecodes file: {err}"))?;
-    bincode::serialize_into(file_bytecodes, &bytecodes)
-        .map_err(|err| format!("Failed to write bytecodes to file: {err}"))?;
+    // Write compressed bytecodes to disk.
+    let serialized = bincode::serialize(&bytecodes)?;
```

Comment:
Maybe something like this would be cleaner?
```rust
let file_bytecodes = File::create("data/bytecodes.bincode.gz")
    .map_err(|err| format!("Failed to create bytecodes file: {err}"))?;
let serialized_bytecodes = bincode::serialize(&bytecodes)
    .map_err(|err| format!("Failed to serialize bytecodes: {err}"))?;
GzEncoder::new(file_bytecodes, Compression::default())
    .write_all(&serialized_bytecodes)
    .map_err(|err| format!("Failed to write bytecodes to file: {err}"))?;
```